### PR TITLE
Aarnq/np kured velero rclone s3export

### LINKED ISSUE #1133: [3]Add NetworkPolicies for velero, s3 exporter, rclone, and kured
This task include creating NetworkPolicies for velero, s3 exporter, rclone, and kured. This includes:
- velero (the whole namespace)
- s3 exporter
- rclone
- kured

This is a subtask to https://github.com/elastisys/compliantkubernetes-apps/issues/654. See that issue for a more detailed instruction, including definition of done.

### PR DESCRIPTION
**Pull request by aarnq**
**What this PR does / why we need it**:
Network policies for Kured, Velero and Restic, rclone-sync and s3-exporter.

**Which issue this PR fixes**:
fixes #1133

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/charts/networkpolicy-common/values.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+fullnameOverride: ""
+nameOverride: ""
+
+global:
+  apiserver:
```
Comment (robinAwallace) [inline helmfile/charts/networkpolicy-common/values.yaml:5]: We have changed to scapiserver and wcapiserver. But i would like to go back to just apiserver. So i could move more common np into common.

Comment (robinAwallace) [reply helmfile/charts/networkpolicy-common/values.yaml:5]: @viktor-f @Xartos :eyes:

Comment (aarnq) [reply helmfile/charts/networkpolicy-common/values.yaml:5]: @robinAwallace Look at the `values/networkpolicies-common.yaml.gotmpl` as well.

Comment (robinAwallace) [reply helmfile/charts/networkpolicy-common/values.yaml:5]: Ah perfect :+1:

File: helmfile/charts/networkpolicy-common/values.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  nodes:
+    ips:
+      - "0.0.0.0/0"
+  objectStorage:
+    ips:
+      - "0.0.0.0/0"
+    ports:
+      - 443
+
+  monitoringSelector:
```
Comment (viktor-f) [inline helmfile/charts/networkpolicy-common/values.yaml:21]: I think this is a bit more accurate
```suggestion
  prometheusSelector:
```

File: helmfile/charts/networkpolicy-common/templates/velero/allow-restic.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  namespace: velero
+spec:
+  policyTypes:
+    - Ingress
+    - Egress
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: velero
+      name: restic
+  egress:
```
Comment (viktor-f) [inline helmfile/charts/networkpolicy-common/templates/velero/allow-restic.yaml:17]: Does velero and restic not talk with each other? Do they just use CRDs?

File: helmfile/charts/networkpolicy-common/values.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  apiserver:
+    ips:
+      - "0.0.0.0/0"
+    port: 6443
+  ingress:
+    ips:
+      - "0.0.0.0/0"
+  nodes:
+    ips:
+      - "0.0.0.0/0"
```
Comment (viktor-f) [inline helmfile/charts/networkpolicy-common/values.yaml:14]: Are these used?

File: helmfile/values/networkpolicy-common.yaml.gotmpl [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  {{- if eq .Environment.Name "service_cluster" }}
+    ips: {{- toYaml .Values.networkPolicies.global.scApiserver.ips | nindent 6 }}
+    port: {{ .Values.networkPolicies.global.scApiserver.port }}
+  {{- else }}
+    ips: {{- toYaml .Values.networkPolicies.global.wcApiserver.ips | nindent 6 }}
+    port: {{ .Values.networkPolicies.global.wcApiserver.port }}
+  {{- end }}
+  objectStorage:
+    ips: {{- toYaml .Values.networkPolicies.global.objectStorage.ips | nindent 6 }}
+    ports: {{- toYaml .Values.networkPolicies.global.objectStorage.ports | nindent 6 }}
```
Comment (viktor-f) [inline helmfile/values/networkpolicy-common.yaml.gotmpl:12]: If we want `nodes` and `ingress` then they should be added here as well.

File: helmfile/values/service-cluster-networkpolicy.yaml.gotmpl [hunk 6]
```diff
@@ -18,7 +18,6 @@ global:
     ports: {{- toYaml .Values.networkPolicies.global.objectStorage.ports | nindent 6 }}
   externalLoadBalancer: {{ .Values.networkPolicies.global.externalLoadBalancer }}
   ingressUsingHostNetwork: {{ .Values.ingressNginx.controller.useHostPort }}
-
 harbor:
```
Comment (viktor-f) [inline helmfile/values/service-cluster-networkpolicy.yaml.gotmpl:21]: Why is this removed? We seem to have a new line between the other components.

File: helmfile/charts/networkpolicy-common/values.yaml (hunk 2 above)
Comment (viktor-f) [reply helmfile/charts/networkpolicy-common/values.yaml:21]: But I really like this idea, it could be adopted in more places.

Comment (aarnq) [reply helmfile/charts/networkpolicy-common/values.yaml:21]: Yeah, a refactor run will be needed after we get all these merged. I think it will be the easiest way to make sure we reuse parts like this.

File: helmfile/charts/networkpolicy-common/values.yaml (hunk 4 above)
Comment (aarnq) [reply helmfile/charts/networkpolicy-common/values.yaml:14]: No, I think they came with it when I copied over the values from either sc or wc. But I'll keep it and add the if/else in the values template so people can build on it.

File: helmfile/values/service-cluster-networkpolicy.yaml.gotmpl (hunk 6 above)
Comment (aarnq) [reply helmfile/values/service-cluster-networkpolicy.yaml.gotmpl:21]: Unintended.

File: helmfile/charts/networkpolicy-common/templates/velero/allow-restic.yaml (hunk 3 above)
Comment (aarnq) [reply helmfile/charts/networkpolicy-common/templates/velero/allow-restic.yaml:17]: Yeah, I had to test this again just to be sure but no they don't talk directly to each other.

**Review by robinAwallace — APPROVED** [review]

**Review by viktor-f — APPROVED** [review]

File: config/config/common-config.yaml [hunk 7]
```diff
@@ -613,3 +613,13 @@ networkPolicies:
     scIngress:
       ips: # ip(s) to sc loadbalancer if available, otherwise sc worker nodes
         - set-me
```
Comment (robinAwallace) [inline config/config/common-config.yaml:615]:
```suggestion
    scIngress:
      ips: # ip(s) to sc loadbalancer if available, otherwise sc worker nodes
        - set-me
     wcIngress:
      # ip(s) to wc loadbalancer if available, otherwise wc worker nodes
      ips: 
        - set-me
```

**Review by Xartos — APPROVED** [review]
